commands/decimal_spiral.go

```diff
@@ -0,0 +1,258 @@
+package commands
+
+import (
+	"errors"
+	"strconv"
+
+    "github.com/bwmarrin/discordgo"
+
+    "github.com/unswpcsoc/PCSocBot/utils"
+)
+
+const(
+    DecimalSpiralLowerLimit = 5
+    DecimalSpiralUpperLimit = 43
+)
+
+var (
+	ErrDecimalSpiralSyntax = errors.New("size is not an integer")
+	ErrDecimalSpiralRange = errors.New("size is not an odd integer between " + strconv.Itoa(DecimalSpiralLowerLimit) + " and " + strconv.Itoa(DecimalSpiralUpperLimit))
+	ErrDecimalSpiralArgs = errors.New("not enough args")
+)
+
+type DecimalSpiral struct {
+	names []string
+	desc string
+}
+
+func NewDecimalSpiral() *DecimalSpiral {
+	return &DecimalSpiral{
+		names: []string{"decimalspiral", "ds"},
+		desc: "Usage: ds <size>. Generate a decimal spiral. Size must be an odd integer between " + strconv.Itoa(DecimalSpiralLowerLimit) + " and " + strconv.Itoa(DecimalSpiralUpperLimit),
+	}
+}
+
+func (d *DecimalSpiral) Names() []string {
+	return d.names
+}
+
+func (d *DecimalSpiral) Desc() string {
+	return d.desc
+}
+
+func (d *DecimalSpiral) Roles() []string {
+	return nil
+}
+
+func (d *DecimalSpiral) Chans() []string {
+	return nil
+}
+
+func (d *DecimalSpiral) MsgHandle(ses *discordgo.Session, msg *discordgo.Message, args []string) (*CommandSend, error) {
+	if len(args) == 0 {
+		return nil, ErrDecimalSpiralArgs
+	}
+
+    size, err := strconv.Atoi(args[0]);
```

Comment:
Go style mandates tabs, do not indent with spaces. I think you can setup gofmt on save in VSCode, something to look into.